### ISSUE TITLE
lua branch very out of date

### ISSUE THREAD

**Issue by wzdev-ci** [Warzone2100/old-trac-import#930]
`keyword_lua` `resolution_fixed` `type_to-do (no feature requests!)`   |    by Kosh

___

Gents / Gals, I have spent a week or so trying to understand the LUA branch, and I am just spinning my wheels.

Is this project still going to use LUA?  There has not been much activity, and my question in ticket #139 does not encourage me that this project is still going to do anything with LUA at all.

The documentation on what needs to be done is poor, the ticket owner doesn't respond to basic questions, and perhaps the most telling sign is that there has not been a update to the LUA branch in ages.

Just what are the goals of switching to LUA from what seems to be the bison/flex tango?  Is it meant to be a 100% replacement for the evil stepsisters, or is it meant to coexist with them?

The forums have practically no useful information about LUA, just that most are waiting for it.  The same goes for the wiki page about LUA, http://developer.wz2100.net/wiki/Lua saying it is lacking documentation on how & what needs to be done, is a understatement.

I have found these files that deal with the old grammar
```
level_lexer.l
audp_lexer.l audp_parser.y
chat_lexer.l chat_parser.y
message_lexer.l message_parser.y
resource_lexer.l resource_parser.y
script_lexer.l script_parser.y
scriptvals_lexer.l scriptvals_parser.y
strres_lexer.l strres_parser.y
```
It appears to me that none of these are modified except for some trivial changes?
They are still used in the build chain, so it appears they are still used by the code base, correct? 

In short, it looks like a dead duck.

I did not think this topic belongs in the forums since it it developer specific.

If work is to continue, may I suggest the developer team merge trunk into lua2 to prevent it from getting data rot?
Update the documentation on how to help in detail!
Remove the unneeded(?) *.l/*.y files in the build chain!

Hopefully someone can answer these questions?

--Kosh

___
_Issue migrated from [trac:930](https://developer.wz2100.net/ticket/930) at 2022-04-15 19:15:05 -0700_
